src/ORM/Search/SearchContextForm.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace SilverStripe\ORM\Search;
+
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Forms\CompositeField;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_FormAction;
+use SilverStripe\Forms\GridField\GridFieldFilterHeader;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\SSViewer;
+
+/**
+ * A form for using a search context in the CMS.
+ *
+ * Note that the form submission goes through to the index action of the controller by default,
+ * and is intended to return the same markup you would ordinarily get for that controller but with
+ * the results filtered appropriately.
+ */
+class SearchContextForm extends Form
+{
+    private SearchContext $searchContext;
+
+    private ?string $searchField = null;
+
+    private static $casting = [
+        'SchemaData' => 'Text',
+        'getSchemaData' => 'Text',
+    ];
+
+    public function __construct(RequestHandler $controller, SearchContext $searchContext, $name = 'SearchForm')
+    {
+        $this->searchContext = $searchContext;
+        $searchFields = clone $searchContext->getSearchFields();
+        $this->setDefaultSearchField($searchFields, $controller);
+        $this->prepareSearchFieldsForCMS($searchFields);
+        parent::__construct($controller, $name, $searchFields);
+        $this->setHTMLID(ClassInfo::shortName($controller) . '_' . $name);
+        $this->disableSecurityToken();
+        $this->setFormMethod('GET');
+        $this->setFormAction($controller->Link());
+        $this->addExtraClass('cms-search-form');
+
+        // Load the form with previously set search data if any
+        $this->loadDataFrom($searchContext->getSearchParams());
+    }
+
+    public function setSearchField(string $fieldName): static
+    {
+        $this->searchField = $fieldName;
+        return $this;
+    }
+
+    public function getSearchField(): string
+    {
+        return $this->searchField;
+    }
+
+    /**
+     * Get the schema for this search context for use in the CMS search filter form.
+     * Note that this is not the same as the form schema (i.e. it includes no information about form fields).
+     */
+    public function getSchemaData(): array
+    {
+        $schemaUrl = $this->getRequestHandler()->Link('schema');
```

Comment:
This could be called `$formSchemaUrl` instead?  OK to keep as `$schemaUrl` if we no longer call the method `getSchemaData()` i.e. remove the word 'Schema' from it

Reply:
Updated the variable name